src/CommonITILObject.php

```diff
@@ -1359,6 +1359,77 @@ protected function handleTemplateFields(array $input)
         return $input;
     }
 
+    protected function manageITILObjectLinkInput($input)
```

Comment:
This method should be tested, and tests should focus on handling of parent/son relations.

Reply:
By the way, the logic contained in this method should probably be moved into `prepareInputForAdd` and `prepareInputForUpdate` on relation classes (`Change_Change`, `Change_Problem`, `Change_Ticket`, ...), so this method would look like:
```php
        if (isset($input['_link'])) {
            $link = $input['_link'];

            if (isset($link['tickets_id_2'])) {
                Toolbox::deprecated();
                $link = [
                    'itemtype_1' => Ticket::class,
                    'items_id_1' => $link['tickets_id_1'] ?? 0,
                    'itemtype_2' => Ticket::class,
                    'items_id_2' => $link['tickets_id_2'],
                    'link'       => $link['link'] ?? CommonITILObject_CommonITILObject::LINK_TO,
                ];
            }

            $link_class = !empty($link['itemtype_1']) && !empty($link['itemtype_2'])
                ? CommonITILObject_CommonITILObject::getLinkClass($link['itemtype_1'], $link['itemtype_2'])
                : null;

            if ($link_class === null) {
                trigger_error(
                    sprintf('Invalid itemtypes "%s"/"%s" on ITIL objects link.', $link['itemtype_1'], $link['itemtype_2']),
                    E_USER_WARNING
                );
                return;
            }

            if ($link['itemtype_1'] == $this->getType() && $link['items_id_1'] == 0) {
                // Link was added in creation form, ID was not available yet
                $link['items_id_1'] = $this->getID();
            }

            $itil_itil = new $link_class();
            if ($itil_itil->add($link)) {
                $input['_forcenotif'] = true;
            } else {
                Session::addMessageAfterRedirect(__('Unknown ITIL Object'), false, ERROR);
            }
        }
```